app/volume.go

```diff
@@ -628,23 +656,30 @@ func (job *Job) shouldDoRecurringBackup(volume *longhornclient.Volume) (bool, er
 
 // getLastBackup return the last backup of the volume
 // return nil, nil if volume doesn't have any backup
-func (job *Job) getLastBackup(volume *longhornclient.Volume) (*longhornclient.Backup, error) {
+func (job *Job) getLastBackup() (*longhornclient.Backup, error) {
+	var err error
+	defer func() {
+		err = errors.Wrapf(err, "failed to get last backup for %v", job.volumeName)
+	}()
+
+	volume, err := job.api.Volume.ById(job.volumeName)
+	if err != nil {
+		return nil, err
+	}
 	if volume.LastBackup == "" {
 		return nil, nil
 	}
-
 	backupVolume, err := job.api.BackupVolume.ById(job.volumeName)
 	if err != nil {
 		return nil, err
 	}
-
 	return job.api.BackupVolume.ActionBackupGet(backupVolume, &longhornclient.BackupInput{
 		Name: volume.LastBackup,
 	})
 }
 
 func (job *Job) listBackupsForCleanup(backups []longhornclient.Backup) []string {
-	sts := []*NameWithTimestamp{}
+	sts := []NameWithTimestamp{}
```

Comment:
moving the elements to the stack seems fine, since it's a small bounded structure.

Reply:
I thought that the slice is still on heap because we do append() that could potentially reallocate the slice so Golang cannot determine the final size of the slice at compile time.